Update user documentation structure

thebe14 opened the pull request:
<!--
A good PR should describe what benefit this brings to the repository.
Ideally, there is an existing issue which the PR address.

Please check the [Contributing guide](https://docs.egi.eu/about/contributing/)
for style requirements and advice.
-->

# Summary

The documentation needs some changes to the structure, in order to accommodate new services and to make it easier to navigate. A lot of (most) content from the EGI-ACE [D2.3](https://docs.google.com/document/d/1ycq076_SQ6pIH3MabruDpCviIERWVQ_l/edit#) and [D2.5](https://docs.google.com/document/d/1ycq076_SQ6pIH3MabruDpCviIERWVQ_l/edit#) should end up in our user documentation.

- Added the necessary new pages
- Provided content for the new pages
- Revised content of existing pages to bring them in line with the rest of the documentation style.
---

<!-- Add, if any, the related issue here, e.g. #6 -->

**Related issue :** Implements #411

andrea-manzi commented:
great work! let's make the linter happy and then we can have a preview and review the changes

gwarf reviewed (commented):
Thanks.
Added a few comments, mainly letting other service owners review the content for their services.
Once it lints well we can enable the preview.

thebe14 commented on content/en/users/compute/orchestration/im/_index.md:
```diff
@@ -0,0 +1,44 @@
+and update, and monitoring of virtual infrastructures. 
+
+IM is integrated with the [EGI Check-in Service](../../../aai/check-in) and supports
+a wide variety of backends, either public (such as
+[Amazon Web Services](https://aws.amazon.com/),
+[Google Cloud](http://cloud.google.com/) or
+[Microsoft Azure](http://azure.microsoft.com/)) or on-premises (such as
+[OpenStack](http://www.openstack.org/)) or federated infrastructures (such as 
+[EGI Cloud Compute](../../cloud-compute)),
+thus making user applications cloud agnostic.
```
Agree.

thebe14 commented on content/en/users/aai/check-in/_index.md:
```diff
@@ -0,0 +1,18 @@
+
+## What is it?
+
+[EGI Check-in](https://www.egi.eu/services/check-in/) is a web service that
```
Updated

thebe14 commented:
> i see one problem now...the links to the doc we have added to the D2.5 deliverables will be incorrect ..so we should revise them ASAP when we approve this PR

EGI-ACE D2.5 updated

gwarf commented:
Looks good!
Some quick/first feedback: I'm not sure at all about https://docs.egi.eu/documentation/414/users/getting-started/task-force/ like about how it's named, where it's in the menu.
The FedCloud Task Force is only a very specific task force, focusing on mainly the cloud activities. I would find this better in the Compute part, or otherwise in something named/organised differently.
Same for the CLI in the getting started, it would like it's a CLI for all EGI services, while it's certainly not the case

thebe14 commented:
> I'm not sure at all about https://docs.egi.eu/documentation/414/users/getting-started/task-force/ like about how it's named, where it's in the menu. The FedCloud Task Force is only a very specific task force, focusing on mainly the cloud activities. I would find this better in the Compute part, or otherwise in something named/organised differently.

That will get improved when the Getting Started part gets revised next. A lot of stuff about how to request access, the architecture, and more from D2.3 and D2.5 need to be added. It is not what this PR should fix. 

> Same for the CLI in the getting started, it would like it's a CLI for all EGI services, while it's certainly not the case

The CLI is more or less "the CLI" for EGI, a lot of docs pages refer to it (there were many places that were introducing fedcloudcli, now all those refer to `/users/getting-started/cli`). Some services have additonal CLIs as well, but the docs for those services will mention them. And fedcloudcli is and will be improved to cover more and more of the EGI services. Finally, improving this is not the purpose of this PR.

thebe14 commented:
@andrea-manzi I did a lot of changes in the top pages for the data services. uniformized titles and subtitles (e.g. no links or acronyms in titles), added a _What is it?_ paragraph to each. please review them, let me know if it's good (or you can improve it further in future PRs).

Data TODOs:

- the openRDM page has to be filled in
- some diagrams from D2.5 could also make their way into the docs

andrea-manzi commented on content/en/users/data/management/_index.md:
```diff
@@ -0,0 +1,62 @@
+  one [Grid](../storage/grid-storage) or [Object](../storage/object-storage) storage
+  to another. It is used internally by [Rucio](rucio) to schedule transfers based
+  on the data policies defined by the users.
+- [openRDM](open-rdm) is a combined [FAIR](https://en.wikipedia.org/wiki/FAIR_data)
```
As we don't have a doc for now i would like the EOSC marketpace entry https://marketplace.eosc-portal.eu/services/openrdm-eu

thebe14 replied:
why not use the 2-3 paragraphs from D2.5? and also link to the eosc marketplace

andrea-manzi commented on content/en/users/security/_index.md:
```diff
@@ -0,0 +1,25 @@
+
+## Using cloud services securely
+
+Cloud security in the EGI Federated Cloud is the highest priority.
```
I think we should not refer only to cloud here, as the service could be used also by other services in the EGI infra

thebe14 replied:
Fixed

andrea-manzi commented on content/en/users/data/management/datahub/_index.md:
```suggestion
{{% alert title="Note" color="info" %}} EGI DataHub is based on the
```

andrea-manzi commented on content/en/users/data/management/datahub/_index.md:
at the the integration with AppDB has never been implemented, so given that we are revising this part i would remove it

thebe14 replied:
Done

enolfc commented on content/en/users/aai/_index.md:
```diff
@@ -0,0 +1,20 @@
+[EGI is a federation](https://www.egi.eu/federation/) of
+compute and storage resource providers.
+
+In order for the providers to be able to
```
I would move this to active instead of passive form

thebe14 replied:
Improved. But I do not know how to rewrite this in active voice, it's too generic.

enolfc commented on content/en/users/aai/check-in/_index.md:
```diff
@@ -0,0 +1,38 @@
+type: docs
+weight: 30
+description: >
+  Control access to resources in the EGI Cloud
```
```suggestion
  Control access to resources in the EGI Infrastructure
```
or something similar

enolfc commented on content/en/users/compute/_index.md:
```diff
@@ -0,0 +1,56 @@
+The compute services are summarized below:
+
+- [Cloud Compute](./cloud-compute/) means **VM-based computing with associated
+  storage**. It delivers a customisable platform where users have complete control
```
```suggestion
  storage**. It delivers a customisable computing resources where users have complete control
```

enolfc commented on content/en/users/compute/_index.md:
```diff
@@ -0,0 +1,56 @@
+  interprocess communication.
+-->
+
+When researchers enounter the need large amounts of computing power, need hybrid
+solutions based on the above mentioned services, or simply do not care to manage the
+underlying compute facilities that will run their workloads,
+[cloud orchestrators](./orchestration/) can help provision, manage, and monitor
+compute resources, which then run scientific workloads.
```
:+1: I like this one

enolfc commented on content/en/users/compute/cloud-compute/automate/_index.md:
```diff
@@ -24,5 +24,5 @@ Automating this activity will help researchers to:
 - Make scientific applications cloud agnostic.
 
-To automate VM deployment, users can use any of the [cloud orchestrators](../../cloud-orchestration/)
+To automate VM deployment, users can use any of the [cloud orchestrators](../../orchestration)
```
```suggestion
To automate VM deployment, you can use any of the [cloud orchestrators](../../orchestration)
```

enolfc commented on content/en/users/compute/cloud-compute/openstack/_index.md:
```diff
@@ -80,7 +80,7 @@ cat /etc/grid-security/certificates/*.pem >> $(python -m requests.certs)
 ## Authentication
```
Consider reorder this file: Put aai before OpenStack

thebe14 replied:
This page I did not touch. All pages under cloud compute that are not showing up in #411 will disappear in a future PR I am preparing.

enolfc commented on content/en/users/compute/orchestration/_index.md:
```diff
@@ -0,0 +1,35 @@
+| Service Name                                 | Workload Type             | Use-Case |
+| -------------------------------------------- | ------------------------- | -------- |
+| [Infrastructure Manager](im)                 | VMs and containers        | Used to run workloads on a single IaaS Cloud provider. |
+| [Elastic Cloud Compute Cluster](ec3)         | VMs and containers        | Used when you need to run workloads on clusters that can be elastically scaled and potentially span more than one IaaS Cloud provider. |
```
Workload Manager missing here. It's probably not in the same category, but still needs to be mentioned

thebe14 replied:
No, you are right. II added it to the table, maybe the description is not the best, though.

enolfc commented on content/en/users/compute/orchestration/ec3/_index.md:
```diff
@@ -0,0 +1,39 @@
+---
+title: Elastic Cloud Compute Clusters
+linkTitle: Cloud Compute Clusters
```
```suggestion
linkTitle: Elastic Cloud Compute Cluster
```

I think we should avoid dissolving the EC3 branding

thebe14 replied:
This is too long for the nav tree. Here users search for "what" not for the official name of a service. We have plenty of space for that in the actual page.

enolfc commented on content/en/users/compute/orchestration/im/cli/_index.md:
```diff
@@ -1,13 +1,12 @@
 ---
-title: "Infrastructure Manager Command-Line Interface"
-linkTitle: "Command Line"
+title: Infrastructure Manager Commands
```
```suggestion
title: Infrastructure Manager command-line interface
```

thebe14 replied:
Fixed

enolfc commented on content/en/users/data/_index.md:
```diff
@@ -0,0 +1,19 @@
+is quite vast and includes both low level
+[storage services](./storage/) and
+[data management](./management/) frameworks, as well as specialized
+[data transfer](./management/data-transfer/) services.
```
Should we move data-transfer to stand outside of data management to keep this logic? @andrea-manzi

andrea-manzi replied:
i would keep like this now, maybe to revisit when also openrdm doc is in

thebe14 replied:
Added content for the openRDM page

enolfc commented on content/en/users/data/management/datahub/links/_index.md:
```diff
@@ -1,9 +1,10 @@
-linkTitle: "Links"
-type: docs
-description: "Related links"
+title: DataHub Links
```
I feel this page is not bringing much, although I foresee similar ones coming for other services as we improve the documentation. For sure I would not call it "Links", maybe something like "Extra information"?

thebe14 replied:
Agree. I left it there, for future improvement. This PR is already doing too much.

enolfc commented on content/en/users/data/management/rucio/_index.md:
```diff
@@ -1,31 +1,30 @@
+allows communities to easily start using and/or test Rucio with little setup
+cost.
 
 ## Official Rucio Pages
```
Going back to the DataHub comment: this is similar to the "links" in DataHub, no? Maybe doing the same for DataHub, move the "Links" content to the overview DataHub page?

thebe14 replied:
Yep. These should be be compacted to the absolute minimum, and the result either put as notes or as links on terms in the text. Future improvement, maybe not in this PR.

enolfc commented on content/en/users/data/management/datahub/_index.md:
```diff
@@ -0,0 +1,118 @@
+---
```
This one is way larger than other services, feels a bit unbalanced. Are we expecting other services to have an overview page that grow this much?

thebe14 replied:
All should be like that. With details in subpages, neatly broken up by subject. So let's keep this, the others we'll improve in future PRs.

enolfc commented on content/en/users/data/storage/_index.md:
```diff
@@ -0,0 +1,39 @@
+
+The differences between Block, Grid, and Object Storage are summarized below:
+
+| Type       | Sharing                                                  | Accounting           | Usage                           |
```
Update this with table 4.1 of D2.5

thebe14 replied:
Nope. Table 4.1 (but not in table form) is on the parent page of this one.

enolfc reviewed (changes requested):
Good to see this @thebe14! 
Some (rather minor) comments here and there.

andrea-manzi commented:
we should remove draft: true from the openRDM page

enolfc commented:
> I agree. That will get improved when the Getting Started part gets revised next. A lot of stuff about how to request access, the architecture, and more from D2.3 and D2.5 need to be added. But it is not what this PR should fix.
> 

Should we just remove the taskforce page in this PR? It's not bringing any value right now and I fear we will be postponing moving it forever. It simply does not belong to the docs.egi.eu but it's more a TCB-like activity and should be documented as such in confluence

thebe14 commented:
> Should we just remove the taskforce page in this PR? It's not bringing any value right now and I fear we will be postponing moving it forever.

Added draft flag, removed links to it.

enolfc reviewed (approved):
LGTM! Thanks @thebe14
